### PR TITLE
Firebase lagging issues

### PR DESCRIPTION
**Pull request by vycius**
* Improved `FirebaseRealtimeApi` with async call and simplified logic;
* All adapters got stable id;

### CODE REVIEW

File: app/src/main/java/com/nmakademija/nmaakademija/MainActivity.java [hunk 1]
```diff
@@ -65,6 +65,7 @@ protected void onCreate(Bundle savedInstanceState) {
 
     void initBottomNavigation() {
         viewPager.setAdapter(new BottomNavigationFragmentPagerAdapter(getSupportFragmentManager()));
+        viewPager.setOffscreenPageLimit(0);
```
Comment (gediminasel) [inline app/src/main/java/com/nmakademija/nmaakademija/MainActivity.java]: I think you cant do that.
`Requested offscreen page limit 0 too small; defaulting to 1`

**Review by gediminasel — CHANGES_REQUESTED** [review]
- Timer doesn't work. I think you need to check 
`if (countDownTimer != null)
      countDownTimer.cancel();` before creating new timer onTimeUntilSessionLoaded.
- `ViewPager.setOffscreenPageLimit(param);` param must be >0

File: app/src/main/java/com/nmakademija/nmaakademija/MainActivity.java (hunk 1 above)
Comment (gediminasel) [inline app/src/main/java/com/nmakademija/nmaakademija/MainActivity.java]: warn : `Requested offscreen page limit 0 too small; defaulting to 1`

**Comment by vycius** [comment]
Fixed both issues

**Review by gediminasel — APPROVED** [review]

**Pull request merged**